app/src/main/java/com/cmput301f20t21/bookfriends/ui/request/RequestActivity.java

```diff
@@ -0,0 +1,177 @@
+            descriptionTextView.setText(book.getDescription());
+            bookStatus.setText(book.getBookStatus().toString());
+        });
+
+        // also getting the image
+        requestViewModel.getImageUri().observe(this, uri -> {
+            Glide.with(this).load(uri).into(bookImage);
+        });
+    }
+
+    /**
+     * display requesters of the book by getting them from FireStore based on bookId
+     * @param bookId
+     */
+    public void displayRequest(String bookId) {
+        requestViewModel.getRequesters(bookId).observe(this, requesters -> {
+            if (requesters != null) {
+                requestDataList.addAll(requesters);
+                requestAdapter.notifyDataSetChanged();
+            }
+        });
+    }
+
+    /**
+     * setup the back button on the title
+     * @param item
+     * @return
+     */
+    @Override
+    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
+        if (item.getItemId() == android.R.id.home) {
+            finish();
+            return true;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    /**
+     * Function to set view by ID, set adapter and build recycler view
+     */
+    public void buildRecyclerView() {
+        recyclerView = findViewById(R.id.request_recycler_view);
+        recyclerView.setHasFixedSize(true);
+        layoutManager = new LinearLayoutManager(this);
+        recyclerView.setLayoutManager(layoutManager);
+        requestAdapter = new RequestAdapter(requestDataList);
+        recyclerView.setAdapter(requestAdapter);
+
+        requestAdapter.setOnItemClickLisener(new RequestAdapter.OnItemClickListener() {
+            @Override
+            public void onRejectClick(int position) {
+                removeItem(position);
+            }
+
+            @Override
+            public void onAcceptClick(int position) {
+                openDialog(position);
+            }
+        });
+    }
+
+    /**
+     * function to remove an item when we click on Reject button
+     * @param position that needs removing
+     */
+    public void removeItem(int position) {
+        Request request = requestDataList.get(position);
+        requestViewModel.removeRequest(request.getId());
+        requestDataList.remove(position);
+        requestAdapter.notifyItemRemoved(position);
+    }
+
+    /**
+     * When user click on the accept button
+     * popup a dialog to prompt user about their action:
+     * accept one item and remove all other items
+     * @param position that we accept the item
+     */
+    public void openDialog(int position) {
+        String acceptedId = requestDataList.get(position).getId();
+        List<String> idsToRemove = new ArrayList<>();
+        for (int i = 0; i < requestDataList.size();i++) {
+            idsToRemove.add(requestDataList.get(i).getId());
+        }
+        idsToRemove.remove(acceptedId);
+        ConfirmDialog confirmDialog = new ConfirmDialog(acceptedId, idsToRemove);
```

Comment:
you are creating a idsToRemove from id and requests here. and it's passed to the confirm dialog, (continued)